core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -115,28 +126,25 @@ public TenantObject setEnabled(final boolean flag) {
 
     /**
      * Gets the subject DN of this tenant's configured trusted
-     * certificate authority.
+     * certificate authority. This method assumes that this tenant is
+     * configured with a single trusted CA.
      * 
+     * @deprecated replaced by {@link #getTrustedCaSubjectNames()}
      * @return The DN or {@code null} if no CA has been set.
      */
+    @Deprecated(since="1.0", forRemoval=true)
     @JsonIgnore
     public X500Principal getTrustedCaSubjectDn() {
-
-        final JsonObject trustedCa = getProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA);
-        if (trustedCa == null) {
-            return null;
-        } else {
-            return Optional.ofNullable((String) getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN))
-                    .map(dn -> new X500Principal(dn)).orElse(null);
-        }
+        final List<X500Principal> subjectdns = Optional.ofNullable(getTrustedCaSubjectNames()).orElse(null);
+        return subjectdns == null ? null : subjectdns.get(0);
     }
 
     /**
-     * Sets the trusted certificate authority to use for authenticating
-     * devices of this tenant.
+     * Sets the trusted anchor for this tenant's Trusted CA using the given public key and subject DN.
```

Comment:
this method should probably be changed to support *adding* a trust anchor instead of setting one, i.e. *addTrustAnchor*